src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -256,6 +256,14 @@ internal BinaryClassificationTrainers(BinaryClassificationCatalog catalog)
                 Evaluate(x.Scores, labelColumnName), x.Scores, x.Fold)).ToArray();
         }
 
+        public BinaryPredictionTransformer<TModel> ChangeModelThreshold<TModel>(BinaryPredictionTransformer<TModel> model, float threshold)
+             where TModel : class
+        {
+            if (model.Threshold == threshold)
+                return model;
+            return new BinaryPredictionTransformer<TModel>(Environment, model.Model, model.TrainSchema, model.FeatureColumnName, threshold, model.ThresholdColumn);
```

Comment:
>model.ThresholdColumn [](start = 140, length = 21)

Technically, Issue #2465 was that we should be able to set the `Threshold` and `ThresholdColumn` properties of the `BinaryPredictionTransformer`. That said, I think that the actual use cases that we care about are changing the `Threshold`; the `ThresholdColumn` seems more important when we are creating a _new_ `BinaryPredictionTransformer`. I actually don't think we need to modify that property. I'll update the issue that just setting a `Threshold` would be nice.

(Note that in practice, we have a `Score` column and a `Probability` column; modulo floating point error, these are a 1:1 mapping, so we will get the same results no matter which one we threshold on. Setting a custom threshold column seems more like a backdoor for crazy things: e.g. creating a custom score based on a few models and / or heuristics and then modifying a `BinaryPredictionTransformer` to score that column instead.)